src/mscorlib/src/Microsoft/Win32/Win32Native.cs

```diff
@@ -1200,7 +1200,7 @@ internal static bool CopyFile(String src, String dst, bool failIfExists)
                   [Out]StringBuilder lpBuffer);
 
         [DllImport(KERNEL32, SetLastError = true, ExactSpelling = true)]
-        internal static extern uint GetCurrentDirectoryW(uint nBufferLength, SafeHandle lpBuffer);
+        internal static extern uint GetCurrentDirectoryW(uint nBufferLength, [Out] char[] lpBuffer);
```

Comment:
Does the ```[Out]``` have any impact here?  I thought char[] was marshaled simply by pinning the array and passing its address, rather than copying any data in either direction.